crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -151,7 +191,26 @@ impl AuthorityStorePruner {
                         } else {
                             error!("Failed to flush objects table");
                         }
-                    }
+                    },
+                    Some((state, callback)) = checkpoint_stream.recv() => {
+                        if !enable_live_pruner {
+                            callback.send(()).expect("failed to notify checkpoint executor");
+                            continue;
+                        }
+                        loop {
+                            match Self::handle_checkpoint(state.clone(), &perpetual_db.objects) {
+                                Ok(pruned) => {
+                                    info!("Pruned {} objects", pruned);
+                                    callback.send(()).expect("failed to notify checkpoint executor");
+                                    break;
+                                }
+                                Err(err) => {
+                                    error!("Failed to prune objects {:?}", err);
```

Comment:
I think pretty soon we should move to tailing the checkpoint table and away from blocking checkpoint execution for object pruning given it is non-critical. Maybe would make sense to add something to the error message like "Failed to prune objects: {:?}, blocking checkpoint execution!"